src/modules/rppt_tensor_effects_augmentations.cpp

```diff
@@ -1004,6 +1004,149 @@ RppStatus rppt_jitter_host(RppPtr_t srcPtr,
     return RPP_SUCCESS;
 }
 
+/******************** fog ********************/
+
+RppStatus rppt_fog_host(RppPtr_t srcPtr,
+                        RpptDescPtr srcDescPtr,
+                        RppPtr_t dstPtr,
+                        RpptDescPtr dstDescPtr,
+                        Rpp32f *gammaTensor,
+                        RpptROIPtr roiTensorPtrSrc,
+                        RpptRoiType roiType,
+                        rppHandle_t rppHandle)
+{
+    RppLayoutParams layoutParams = get_layout_params(srcDescPtr->layout, srcDescPtr->c);
+    Rpp32f *fogAlphaMaskPtr_1920_1080, *fogIntensityMaskPtr_1920_1080;
+    fogAlphaMaskPtr_1920_1080 = &fogAlphaMask_1920_1080[0];
+    fogIntensityMaskPtr_1920_1080 = &fogIntensityMask_1920_1080[0];
+    Rpp32u height = srcDescPtr->h;
+    Rpp32u width = srcDescPtr->w;
+
+    RpptLayout layout = NCHW;
+    RppLayoutParams srcLayoutParams = get_layout_params(layout, 1);
+    RpptDesc fogMaskSrcDesc ;
+    RpptDescPtr fogMaskSrcDescPtr = &fogMaskSrcDesc;
+    fogMaskSrcDescPtr->numDims = 3;
+    fogMaskSrcDescPtr->offsetInBytes = 0;
+    fogMaskSrcDescPtr->dataType = RpptDataType::F32;  
+    fogMaskSrcDescPtr->n = 1;
+    fogMaskSrcDescPtr->c = 1;
+    fogMaskSrcDescPtr->h = FOG_MAX_HEIGHT;
+    fogMaskSrcDescPtr->w = FOG_MAX_WIDTH; 
+    fogMaskSrcDescPtr->strides.nStride = FOG_MAX_WIDTH * FOG_MAX_HEIGHT;
+    fogMaskSrcDescPtr->strides.hStride = FOG_MAX_WIDTH;
+    fogMaskSrcDescPtr->strides.wStride = 1;
+    fogMaskSrcDescPtr->strides.cStride = FOG_MAX_WIDTH * FOG_MAX_HEIGHT;
+    fogMaskSrcDescPtr->layout = NCHW;
+
+    RpptDesc fogMaskDstDesc ;
+    RpptDescPtr fogMaskDstDescPtr = &fogMaskDstDesc;
+    fogMaskDstDescPtr->numDims = 3;
+    fogMaskDstDescPtr->offsetInBytes = 0;
+    fogMaskDstDescPtr->dataType = RpptDataType::F32;  
+    fogMaskDstDescPtr->n = 1;
+    fogMaskDstDescPtr->c = 1;
+    fogMaskDstDescPtr->h = height;
+    fogMaskDstDescPtr->w = width;
+    fogMaskDstDescPtr->strides.nStride = height * width;
+    fogMaskDstDescPtr->strides.hStride = width;
+    fogMaskDstDescPtr->strides.wStride = 1;
+    fogMaskDstDescPtr->strides.cStride = height * width;
+    fogMaskDstDescPtr->layout = NCHW;
+
+
```

Comment:
Please add a one liner comments for each section so that is is easier to understand